### PR TITLE
update: Updating to Miden v0.4 and concurrent runs

### PR DESCRIPTION
**Pull request by Dominik1999**
I am updating to Miden v0.4 and changing it to include the breaking changes.

### CODE REVIEW

File: benchmarking-cli/src/main.rs [hunk 1]
```diff
@@ -87,46 +86,52 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
...
+
     let now = Instant::now();
     let _program2 = assembler
         .compile(&program_string)
         .expect("Could not compile source");
 
     println! {"Compilation Time (hot): {} ms", now.elapsed().as_millis()}
 
+
+    let stack_input_cloned = stack_input.clone();
```
Comment (Dominik1999) [inline benchmarking-cli/src/main.rs:120]: I am unsure if I need to `clone()` those. In Miden v0.3 I was able to borrow but that seems not possible anymore.

File: playground/src/pages/CodingEnvironment.tsx [hunk 2]
```diff
@@ -91,10 +91,12 @@ an advice_tape.`;
 
       return false;
     }
+
     Object.keys(jsonInput).forEach((key) => {
```
Comment (Dominik1999) [inline playground/src/pages/CodingEnvironment.tsx:95]: Added this error fix - which has nothing to do with the Updage to Miden v0.4. Just saw it.

File: benchmarking-cli/src/main.rs (hunk 1 above)
Comment (hackaugusto) [reply benchmarking-cli/src/main.rs:122]: It seems the API changed and now a clone is required.

**Review by hackaugusto — APPROVED** [review]
LGTM

**Pull request merged**